src/main/java/com/esri/samples/scene/get_elevation_at_a_point/GetElevationAtAPointSample.java

```diff
@@ -0,0 +1,179 @@
+
+            // create a graphics overlay
+            graphicsOverlay = new GraphicsOverlay(GraphicsOverlay.RenderingMode.DYNAMIC);
+            graphicsOverlay.getSceneProperties().setSurfacePlacement(LayerSceneProperties.SurfacePlacement.RELATIVE);
+            sceneView.getGraphicsOverlays().add(graphicsOverlay);
+
+            // add a camera and initial camera position
+            Camera camera = new Camera(28.42, 83.9, 10000.0, 10.0, 80.0, 0.0);
+            sceneView.setViewpointCamera(camera);
+
+            // create listener to handle clicked
+            sceneView.setOnMouseClicked(event -> {
+
+                // get the clicked screenPoint
+                Point2D screenPoint = new Point2D(event.getX(), event.getY());
+                // convert the screen point to a point on the surface
+                Point relativeSurfacePoint = sceneView.screenToBaseSurface(screenPoint);
+
+                // check that the point is on the surface, and primary button was clicked
+                if (relativeSurfacePoint != null && event.getButton() == MouseButton.PRIMARY && event.isStillSincePress()) {
+
+                    // clear any existing graphics from the graphics overlay
+                    graphicsOverlay.getGraphics().clear();
+
+                    // construct a polyline to use as a marker
+                    PolylineBuilder polylineBuilder = new PolylineBuilder(relativeSurfacePoint.getSpatialReference());
+                    Point baseOfPolyline = new Point(relativeSurfacePoint.getX(), relativeSurfacePoint.getY(), 0);
+                    polylineBuilder.addPoint(baseOfPolyline);
+                    Point topOfPolyline = new Point(baseOfPolyline.getX(), baseOfPolyline.getY(), 750);
+                    polylineBuilder.addPoint(topOfPolyline);
+                    Polyline markerPolyline = polylineBuilder.toGeometry();
+                    Graphic polylineGraphic = new Graphic(markerPolyline, lineSymbol);
+                    graphicsOverlay.getGraphics().add(polylineGraphic);
+
+                    // create a point to place the elevationTextSymbol above the polyline
+                    Point textSymbolPosition = new Point(topOfPolyline.getX(), topOfPolyline.getY(), topOfPolyline.getZ()+100);
+
+                    // get the surface elevation at the surface point
+                    ListenableFuture<Double> elevationFuture = scene.getBaseSurface().getElevationAsync(relativeSurfacePoint);
+                    elevationFuture.addDoneListener(() -> {
+                            try {
+                                // get the surface elevation
+                                Double elevation = elevationFuture.get();
+
+                                // update the text in the elevation marker
+                                updateMarkerText(elevationTextSymbol,textSymbolPosition,elevation);
```

Comment:
You don't need a separate method for this. Generally, you only need a separate method if you're going to reuse it from multiple places.

Reply:
Done. I thought having the code there will distract a bit from the essence of the sample (`scene.getBaseSurface().getElevationAsync(relativeSurfacePoint);`), but I agree that it is not being reused.